### PR TITLE
Use archive creation time as mtime for FUSE mount (fixes #2834)

### LINKED ISSUE #2834: Modification time in fuse mount should show creation time of archive
STR: (tested with 1.1.0b6)
``` 
# create some backups
borg mount :: /tmp/mount
ls -la /tmp/mount
total 0
drwxr-xr-x.  1 rugk rugk   0 Jul 16 18:58 .
drwxrwxrwt. 27 root root 660 Jul 16 18:58 ..
drwxr-xr-x.  1 rugk rugk   0 Jul 16 18:58 backup-block-device
drwxr-xr-x.  1 rugk rugk   0 Jul 16 18:58 backup1
drwxr-xr-x.  1 rugk rugk   0 Jul 16 18:58 backup2
drwxr-xr-x.  1 rugk rugk   0 Jul 16 18:58 backup3
drwxr-xr-x.  1 rugk rugk   0 Jul 16 18:58 rugk-2017-07-16T18:51:34
drwxr-xr-x.  1 rugk rugk   0 Jul 16 18:58 rugk-2017-07-16T18:52:19
```

They are all set to the current time. It would be better if the directories show the creation time of the archives as their "modification time".

### PR DESCRIPTION
**Pull request by milkey-mouse**
This is already stored in the `ArchiveInfo` from the manifest so it can still lazy-load the archive itself.

### CODE REVIEW

File: src/borg/fuse.py [hunk 1]
```diff
@@ -242,16 +243,20 @@ def _create_filesystem(self):
...
                     # process archives immediately
-                    self.process_archive(archive_name)
+                    self.process_archive(archive.name)
                 else:
                     # lazily load archives, create archive placeholder inode
-                    archive_inode = self._create_dir(parent=1)
-                    self.contents[1][os.fsencode(archive_name)] = archive_inode
-                    self.pending_archives[archive_inode] = archive_name
+                    archive_inode = self.allocate_inode()
+                    self.items[archive_inode] = deepcopy(self.default_dir)
```
Comment (enkore) [inline src/borg/fuse.py]: `Item(**self.default_dir.as_dict())` is perhaps less magic

File: src/borg/fuse.py [hunk 2]
```diff
@@ -242,16 +243,20 @@ def _create_filesystem(self):
...
+                    created_ts = int(archive.ts.timestamp() * 1e9)
+                    self.items[archive_inode].mtime = created_ts
```
Comment (enkore) [inline src/borg/fuse.py]: It is a bit more complicated, search this file for `have_fuse_xtime_ns` too see how integer-nanoseconds are exposed to FUSE.

Comment (milkey-mouse) [inline src/borg/fuse.py]: Does that need to be accounted for here? The `have_fuse_xtime_ns` bit seems to always use `item.mtime` in microseconds.

File: src/borg/fuse.py [hunk 3]
```diff
@@ -242,16 +242,18 @@ def _create_filesystem(self):
...
+                    self.process_archive(archive.name)
                 else:
                     # lazily load archives, create archive placeholder inode
-                    archive_inode = self._create_dir(parent=1)
-                    self.contents[1][os.fsencode(archive_name)] = archive_inode
-                    self.pending_archives[archive_inode] = archive_name
+                    archive_inode = self.allocate_inode()
+                    self.items[archive_inode] = Item(**self.default_dir.as_dict())
+                    self.items[archive_inode].mtime = int(archive.ts.timestamp() * 1e9)
+                    self.parent[archive_inode] = 1
```
Comment (ThomasWaldmann) [inline src/borg/fuse.py]: hmm, maybe add a mtime=None param to _create_dir and move the code there?

File: src/borg/fuse.py [hunk 4]
```diff
@@ -242,16 +242,15 @@ def _create_filesystem(self):
...
                 if self.versions:
                     # process archives immediately
-                    self.process_archive(archive_name)
+                    self.process_archive(archive.name)
                 else:
                     # lazily load archives, create archive placeholder inode
-                    archive_inode = self._create_dir(parent=1)
-                    self.contents[1][os.fsencode(archive_name)] = archive_inode
-                    self.pending_archives[archive_inode] = archive_name
+                    archive_inode = self._create_dir(parent=1, mtime=int(archive.ts.timestamp() * 1e9))
```
Comment (ThomasWaldmann) [inline src/borg/fuse.py]: move the computation into _create_dir, give the float timestamp as param (or rename mtime param to mtime_ns and give the ns integer value).

there, you need to consider the value of have_fuse_xtime_ns to compute the correct value and choose the correct attribute name to assign that value. see the other place where have_fuse_xtime_ns is used.

Comment (milkey-mouse) [inline src/borg/fuse.py]: I see the `have_fuse_xtime_ns` code, but doesn't that run for the `EntryAttributes` object and convert it there? Why do I need to convert it if it's not necessary for [`self.default_dir`](https://github.com/borgbackup/borg/blob/master/src/borg/fuse.py#L232) here?

Comment (ThomasWaldmann) [inline src/borg/fuse.py]: I have a look right now...

Comment (ThomasWaldmann) [inline src/borg/fuse.py]: You're right. we are constructing an Item here, which has mtime in nanoseconds always.

Only the entry constructed by FUSE's getattr needs to consider the have_fuse_xtime_ns flag.

OK, so we keep it as is.

File: src/borg/fuse.py [hunk 5]
```diff
@@ -9,6 +9,7 @@
 from collections import defaultdict
 from signal import SIGINT
 from distutils.version import LooseVersion
+from copy import deepcopy
```
Comment (ThomasWaldmann) [inline src/borg/fuse.py]: not used any more?

**Comment by milkey-mouse** [comment]
sure

**Comment by milkey-mouse** [comment]
oops...

**Pull request merged**